config/cmake/addBoost.cmake

```diff
@@ -201,7 +201,7 @@ else()
 	endforeach()
 endif()
 
-set_target_properties(Boostlibs::core Boostlibs::test PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES ${Boost_INCLUDE_DIR})
+set_target_properties(Boostlibs::core Boostlibs::test PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${Boost_INCLUDE_DIR})
```

Comment:
I think we wanted boost libraries marked as system libraries so they don't produce warnings at some warning levels.    Why was this changed?

Reply:
It's another "both of the above". `INTERFACE_SYSTEM_INCLUDE_DIRECTORIES` is transitive, but only marks an include directory as system, it doesn't do an `-isystem` by itself. Fixing this now